hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/util/PlatformName.java

```diff
@@ -45,9 +45,34 @@ public class PlatformName {
 
   /**
    * A public static variable to indicate the current java vendor is
-   * IBM java or not.
+   * IBM and the type is Java Technology Edition which provides its
+   * own implementations of many security packages and Cipher suites.
+   * Note that these are not provided in Semeru runtimes:
+   * See https://developer.ibm.com/languages/java/semeru-runtimes/
    */
-  public static final boolean IBM_JAVA = JAVA_VENDOR_NAME.contains("IBM");
+  public static final boolean USE_IBM_JAVA_PACKAGES = shouldUseIbmPackages();
+
+  /*
+  * IBM_JAVA must be preserved due to the public nature of the property.
+  */
+  @Deprecated
+  public static final boolean IBM_JAVA = USE_IBM_JAVA_PACKAGES;
```

Comment:
can't you just leave this as is and so not need to change so many other files?

Reply:
Thanks you for the review @steveloughran - I have pushed up another commit implementing this suggestion, I am happy to keep it as is - my initial thoughts were that the symbols meaning has changed, but the adjustments in the comments should be sufficient to keep it clear whilst not changing as many files or requiring the deprecation.